tests/test_templatetags_component.py

```diff
@@ -488,75 +509,149 @@ def test_raises_on_invalid_args(self):
 
 
 class MultiComponentTests(BaseTestCase):
-    def register_components(self):
+    @parametrize_context_behavior(["django", "isolated"])
+    def test_both_components_render_correctly_with_no_slots(self):
         registry.register("first_component", SlottedComponent)
         registry.register("second_component", SlottedComponentWithContext)
 
-    def make_template(self, first_slot: str = "", second_slot: str = "") -> Template:
-        template_str: types.django_html = f"""
-            {{% load component_tags %}}
-            {{% component 'first_component' %}}
-                {first_slot}
-            {{% endcomponent %}}
-            {{% component 'second_component' variable='xyz' %}}
-                {second_slot}
-            {{% endcomponent %}}
+        template_str: types.django_html = """
+            {% load component_tags %}
+            {% component 'first_component' %}
+            {% endcomponent %}
+            {% component 'second_component' variable='xyz' %}
+            {% endcomponent %}
         """
-        return Template(template_str)
+        template = Template(template_str)
+        rendered = template.render(Context())
 
-    def expected_result(self, first_slot: str = "", second_slot: str = "") -> str:
-        first_slot = first_slot or "Default header"
-        second_slot = second_slot or "Default header"
-        return f"""
-            <custom-template>
-                <header>{first_slot}</header>
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <custom-template data-djc-id-a1bc40>
+                <header>
+                    Default header
+                </header>
                 <main>Default main</main>
                 <footer>Default footer</footer>
             </custom-template>
-            <custom-template>
-                <header>{second_slot}</header>
+            <custom-template data-djc-id-a1bc44>
+                <header>
+                    Default header
+                </header>
                 <main>Default main</main>
                 <footer>Default footer</footer>
             </custom-template>
-        """
-
-    def wrap_with_slot_tags(self, s):
```

Comment:
I refactored these tests to remove the methods like `wrap_with_slot_tags` here, and instead inlined the code into the tests. So the tests are now dumber and more verbose, but easier to read.